src/main/java/io/kuzzle/sdk/core/Kuzzle.java

```diff
@@ -1426,52 +1342,160 @@ private void reconnect() {
+  protected WebSocketClient createSocket() throws URISyntaxException {
+    URI uri = null;
+    try {
+      uri = new URI("ws"+(this.isSsl ? "s" : "")+"://"+this.host+":"+this.port+"/");
+    }
+    catch (URISyntaxException e) {
+      e.printStackTrace();
+    }
+    socket = new WebSocketClient(uri) {
+      @Override
+      public void onOpen() {
+        if (Kuzzle.this.state == States.OFFLINE) { // Reconnect
+          Kuzzle.this.state = States.CONNECTED;
+
+          if (Kuzzle.this.jwtToken != null) {
+            Kuzzle.this.checkToken(jwtToken, new ResponseListener<TokenValidity>() {
+              @Override
+              public void onSuccess(TokenValidity response) {
+                if (!response.isValid()) {
+                  Kuzzle.this.jwtToken = null;
+                  Kuzzle.this.emitEvent(Event.tokenExpired);
+                }
+
+                Kuzzle.this.reconnect();
+              }
+
+              @Override
+              public void onError(JSONObject error) {
+                Kuzzle.this.jwtToken = null;
+                Kuzzle.this.emitEvent(Event.tokenExpired);
+                Kuzzle.this.reconnect();
+              }
+            });
+          } else {
+            Kuzzle.this.reconnect();
+          }
+        } else {
+          Kuzzle.this.state = States.CONNECTED;
+
+          Kuzzle.this.renewSubscriptions();
+          Kuzzle.this.dequeue();
+          Kuzzle.this.emitEvent(Event.connected);
+
+          if (Kuzzle.this.connectionCallback != null) {
+            Kuzzle.this.connectionCallback.onSuccess(null);
+          }
+        }
+      }
+
+      @Override
+      public void onTextReceived(String message) {
+        try {
+          JSONObject json = new JSONObject(message);
+          OnQueryDoneListener listener = null;
+          if (json.has("requestId")) {
+            listener = currentQueries.get(json.getString("requestId"));
+          } else {
+            listener = currentQueries.get(json.getString("room"));
+          }
+
+          if (listener != null) {
+            // checking token expiration
+            if (!json.isNull("error") && json.getJSONObject("error").getString("message").equals("Token expired") && !json.getString("action").equals("logout")) {
+              emitEvent(Event.tokenExpired, listener);
+            }
+
+            if (!json.isNull("error")) {
+              listener.onError(json.getJSONObject("error"));
+            } else {
+              listener.onSuccess(json);
+            }
+            currentQueries.remove(json.getString("requestId"));
+          }
+
+          EventListener l  = roomList.get(json.getString("room"));
+          if (l != null) {
+            l.trigger(new JSONObject(message));
+          }
+        } catch (JSONException e) {
+          throw new RuntimeException(e);
+        }
+      }
+
+      @Override
+      public void onBinaryReceived(byte[] data) {
+
+      }
+
+      @Override
+      public void onPingReceived(byte[] data) {
```

Comment:
The websocket class handles ping/pong packets. I don't think you should override those methods as this is the same as ignoring ping frames.